src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy

```diff
@@ -78,11 +79,13 @@ final class OxfordNanoporeMeasurement {
     }
 
     private void assessPooledStatus() {
-        this.pooledSamplesMeasurement = folders["fast5pass"] ? folders["fast5pass"].getChildren().get(0) instanceof Fast5Folder : false
+        this.pooledSamplesMeasurement = folders["fast5pass"]?.getChildren() ?
+            folders["fast5pass"].getChildren().get(0) instanceof Fast5Folder : false
         // There can be still pooled samples in the failed folder, worst case is all
         // samples failed, so we need to check there to
         if (! pooledSamplesMeasurement) {
-            this.pooledSamplesMeasurement = folders["fast5fail"] ? folders["fast5fail"].getChildren().get(0) instanceof Fast5Folder : false
+            this.pooledSamplesMeasurement = folders["fast5fail"]?.getChildren() ? folders["fast5fail"]
+                .getChildren().get(0) instanceof Fast5Folder : false
```

Comment:
This is a bit too crazy for me without comments :)

Reply:
sure, i will make it more declarative

Reply:
I think this is fine. It first checks if there are any children of the fast5fail and then assigns the first child as a Fast5Folder. In case there are no children of the fast5fail it assigns false

Reply:
I think the fact that you missed the first check:
`folders["fast5fail"]?`
shows that it isn't that easy to read. This is giving me Functional Programming PTSD.